Several Weeks Worth of Changes

dangoh123 opened the pull request:
Committing several weeks worth of commits to my fork, including the iron line plots, and several changes organisation and to the way that animations and renders are organised.

The current thing we are working on that is causing us to get stuck is to do with getting our arbitrary `f` function to go onto a geodesic render. The way that our new `f` function works is to find the stable orbits for a preset `r_vals`, and give them a corresponding `f_vals`, and calculate the plot and temperature values from there pretty much as before. I'm not too sure how to implement this into our main temperature render, as this uses the value (point?) functions, which seem to rely on an analytical function that can compute a temperature at any given radius, rather than a precomputed "lookup table" (I'm not sure if that's the best term to refer to it, but it's what we've been saying) of values. I've tried adding additional arguments to value functions before, but have never been able to get this working. Do you know of any good way to do this?

dangoh123 committed:
Fixed merge issues

Previous commit, hopefully not overriding changes to `redshift.jl`

dangoh123 committed:
Removed Plots import.

Removed Plots import from emission_profile.jl

dangoh123 committed:
Merge branch 'main' of https://github.com/dangoh123/AccretionFormulae.jl

dangoh123 committed:
Moved and Refactored Temperature Render, and Added Animations

Moved all test scripts to a new scripts folder. Refactored the code in `temperature_render.jl` to be in a function, with default arguments that can be quickly changed to modify the properties of the render.

This is used with a test animation script which can generate animations, varying parameters such as the observation angle.

dangoh123 committed:
Made Changes to Line Profile

dangoh123 committed:
Resolved conflict with old version

dangoh123 committed:
Renamed, Cleaned Up, and Commented Scripts

Went through the scripts used to generate various pictures for the
poster and cleaned them up, and also renamed them to more sensible
names.

Have not updated code for the new caching changes, as I wanted to commit
these old versions first.

dangoh123 committed:
Syncing With Origin

Trying to remove outdated files from origin

dangoh123 committed:
Merge branch 'main' of https://github.com/dangoh123/AccretionFormulae.jl

dangoh123 committed:
Fetching Upstream and Merging Conflicts

dangoh123 committed:
Normalised Peaks to Same Value

dangoh123 committed:
Testing Histogram

dangoh123 committed:
Changed Multiplier

dangoh123 committed:
Should Save Images

dangoh123 committed:
Testing New Line Profile on Solva

dangoh123 committed:
Plots For Solva

dangoh123 committed:
Changes for solva

dangoh123 committed:
Saving Images

dangoh123 committed:
Legend Changes

dangoh123 committed:
Changing Animations For Solva

dangoh123 committed:
Testing Combined Images

Testing running a combined render and iron line profile animation with a
changing observation angle to be run on solva.

dangoh123 committed:
Fixed Previous Commit

dangoh123 committed:
Small Changes to Animation

dangoh123 committed:
Fixed previous commit

dangoh123 committed:
Fixed previous commit again

dangoh123 committed:
Added Specific Animations

dangoh123 committed:
Changes for animation

dangoh123 committed:
Small Script Parameter Changes

dangoh123 committed:
Added Test for Circular Geodesics

Added a new script that searches through different combinations of
initial radius, and the map-impact parameters alpha and beta to find the
combinations which form stable circular orbits.

dangoh123 committed:
Added a new version of line profile

Used the ideas from Andy and Fergus' methods to make our own method,
which seems to be working well so far, using the cache generated by a
heatmap.

dangoh123 committed:
Various modifications on laptop

Moving several files to continue the work on desktop.

dangoh123 committed:
Organised Files and Modified Combined Render

Moved scripts into appropriate folders for their needed purpose, and
created and updated version of the combined render, to show the effects
of observation angle.

dangoh123 committed:
Added some unsaved changes

dangoh123 committed:
Added an iron line plot for varying angles

dangoh123 committed:
Seperated spin and angle plots

dangoh123 committed:
Changed image saving filenames

dangoh123 committed:
Small aesthetic changes

dangoh123 committed:
using Revise
using GeodesicTracer
using ComputedGeodesicEquations
using StaticArrays
using Plots
using AccretionFormulae
using Dates
gr()

init_radii = 10:1.0:100
vϕ_vals = 3.00:0.01:10.00
total = length(init_radii)*length(vϕ_vals) # total number of combinations
found = 0   # number of combinations found

start = now() # timing start
for init_radius in init_radii

    m = BoyerLindquist(M=1.0, a=1.0)
    R_isco = AccretionFormulae.r_isco(m.a, m.M)
    u = @SVector [0.0, init_radius, deg2rad(90.0), 0.0]

    # bypass map_impact_parameters cus it's not ideal for this
    # still have to brute force search this bit
    # or use some analytic result from a paper
    # vs = [@SVector [0.0, 0.0, 0.0, -vϕ/100] for vϕ in 3.96:0.01:3.97]
    vs = [@SVector [0.0, 0.0, 0.0, -vϕ/100] for vϕ in vϕ_vals]
    us = [u for _ in 1:length(vs)]

    sols = tracegeodesics(
        m,
        us, vs,
        (0.0, 300.0),
        μ=1.0;
        abstol=1e-8,
        reltol=1e-8,
        verbose=true
    )
    # @show(typeof(sols))
    for (i, sol) in enumerate(sols)
        # selecting only radius values, and only looking at the end
        new_sols = copy(selectdim(sol, 1, 6))
        # @show(new_sols)
        min, max =  extrema(new_sols)
        diff = max - min

        # if the difference is small enough, and it doesn't crash into the ISCO
        # it is a valid solution, and is displayed
        if (diff <= 1) & (min > R_isco*1.1)
            found +=1
            plt = plot(sol, vars=(8, 6), projection=:polar, range=(0, 21), legend=false)
            title!("Initial Radius=$init_radius, vϕ=$(vϕ_vals[i])")
            display(plt)
        end
    end
end
duration = now() - start # timing finish
println("Done")
println("Found $found solutions in $duration.")
println("Success rate of $(found/total) .")

dangoh123 committed:
Updated geodesic search to vary duration based on radius

dangoh123 committed:
Added heatmap for iron lines at different parameters

dangoh123 committed:
Small update to iron_line_profile

dangoh123 committed:
Small changes to run on solva

dangoh123 committed:
Adding files to continue on laptop

dangoh123 committed:
Added unsaved changes

dangoh123 committed:
Updated Emissivity Deivations

dangoh123 committed:
Small Changes

dangoh123 committed:
Added an emission profile for an arbitrary metric

Works by finding geodesics, then using eq 27 from Johannsen (2014).
Works to a reasonable degree so far, but has negative values, and is
quite noisy and slow.

dangoh123 committed:
Added a temperature profile to the aribtrary f

dangoh123 committed:
Updated Scripts For Aribtrary Profile

dangoh123 committed:
Resolved Conflicts

fjebaker commented:
There's a lot of files in this commit -- if you have output of what they do, would you mind just posting them to this PR so we can keep track of things?

As to your issue with the tying in the circular orbit `f` with disc tracing -- I have a few ideas for this, but it may be a little bit of work.

Have you been able to verify that the `f` you calculate with the circular orbits is the same as the `f` in the Page & Thorne paper?

fjebaker commented:
Principally, the key observation here is that you should be able to find a stable orbit for any `r > r_ISCO`; therefore, a recipe is

1. trace observer to disc to find intersections with the disc,
2. for each unique annulus (this could be binned in a lookup-table as you say, or otherwise), find the velocity of the stable circular orbit, and then
3. use this to compute f at that location.

The easiest way to do this is just to write point functions that extracts the radius, find stable orbits for each `r+dr`, use those to compute `f` in that interval, and then use those `f`s to redraw the render using some sort of substitution rule.

I have something that may help you with all of this, which I can hopefully send you later today, but I have a few other priorities to finish first...

dangoh123 committed:
Added Working f Function Approximations

Added a fully working integration of the f function using the numerical
approximation method that Fergus helped devlop, and with the equation
from Johannsen (2014). Allows you to change the values of epsilon, and
the alphas, which have an effect on the f function, which will change
the renders. The iron line plots are roughly the same, even with large
changes which is quite an interesting observations to note.

dangoh123 committed:
Fixed Iron Line Profile Not Using Correct Flux

dangoh123 committed:
Added some renders to be used on Solva

dangoh123 committed:
Moved File

dangoh123 committed:
Added Solva Plots

dangoh123 committed:
Changed Values

dangoh123 committed:
Changed Heatmap

dangoh123 committed:
Changed Heatmap

dangoh123 committed:
Added Arbitrary Renders for Solva

dangoh123 committed:
Modified renders

dangoh123 committed:
Fixed Arbitrary Temperature Renders

dangoh123 committed:
Added file to run all renders at once on Solva

dangoh123 committed:
Fixed Settings

dangoh123 committed:
Fixed Iron Line Heatmap

dangoh123 committed:
Changed files to run

dangoh123 committed:
Fixed File Location

dangoh123 committed:
Changed files to run

dangoh123 committed:
Fixed Heatmap Settings

dangoh123 committed:
Modified Combined Render

dangoh123 committed:
Changed Parameters

dangoh123 committed:
Execution Time Test

dangoh123 committed:
Unsaved Changes

dangoh123 committed:
Changed Units on ISCO Plot

dangoh123 committed:
Fixed Last Commit

dangoh123 committed:
Changed Labels

dangoh123 committed:
Changed Labels

dangoh123 committed:
Changes to run on solva

dangoh123 committed:
Fixed Syntax Error

dangoh123 committed:
Showing Results

dangoh123 committed:
Fixed Last Commit

dangoh123 committed:
Fixed Error

dangoh123 committed:
Changed parameters

dangoh123 committed:
Stuff on laptop

dangoh123 committed:
Stuff on laptop
